src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -318,15 +539,71 @@ Eigen::VectorXd RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::reduceVector(
 template <typename RADIAL_BASIS_FUNCTION_T>
 void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshFirstRound()
 {
-  PRECICE_CHECK(not utils::MasterSlave::isSlave() && not utils::MasterSlave::isMaster(),
-                "RBF mapping is not supported for a participant in master mode, use petrbf instead");
+  PRECICE_TRACE();
+  mesh::PtrMesh filterMesh, otherMesh;
+  if (getConstraint() == CONSISTENT) {
+    filterMesh = input();  // remote
+    otherMesh  = output(); // local
+  } else if (getConstraint() == CONSERVATIVE) {
+    filterMesh = output(); // remote
+    otherMesh  = input();  // local
+  }
+
+  if (otherMesh->vertices().empty())
+    return; // Ranks not at the interface should never hold interface vertices
+
+  // Tags all vertices that are inside otherMesh's bounding box, enlarged by the support radius
+
+  if (_basisFunction.hasCompactSupport()) {
+
+    auto rtree    = mesh::rtree::getVertexRTree(filterMesh);
+    namespace bgi = boost::geometry::index;
+    auto bb       = otherMesh->getBoundingBox();
+    // Enlarge by support radius
+    bb.expandBy(_basisFunction.getSupportRadius());
+    rtree->query(bgi::satisfies([&](size_t const i) { return bb.contains(filterMesh->vertices()[i]); }),
```

Comment:
The spatial rtree can execute an intersection query faster than an evaluating a predicate, but this is out of scope. I'll open an issue.

Reply:
Opened the issue: #762